src/Shelly1/shelly_init.cpp

```diff
@@ -54,10 +64,28 @@ void CreateComponents(std::vector<std::unique_ptr<Component>> *comps,
     comps->emplace_back(std::move(gdo));
     return;
   }
-  // Single switch with non-detached input = only one accessory.
-  bool to_pri_acc = (mgos_sys_config_get_sw1_in_mode() != 3);
+
+  // Sensor Discovery
+  std::vector<std::unique_ptr<TempSensor>> sensors;
+  onewire->DiscoverAll(NUM_SENSORS_MAX, &sensors);
```

Comment:
i have a request: if no sensors are found during discovery, can you please return GPIO0 to input?
the reason being, in debugging setup with serial cable connected, GPIO0 is connected to DTR line of the serial adapter, which is an output, driving it high. having two push-pull outputs connected to one another is not a good idea.
having it only briefly for detection is fine, but i don't want to leave it for long time. so, if nothing is found, you should destroy the onewire object and return gpio0 to input with pull-up.

Reply:
this turns out a little harder as expected: when deleting the onewire instance, the code as is would not find the sensors on a restart of the `CreateComponents`, as it is only initialized on `CreatePeripherals`.

this leads to the question: should the sensors already be discovered on `CreatePeripherals` as well?
or should the onewire object persist and then we claim/free the pin on every `CreateComponents`

Reply:
right now i moved everything to `CreateComponents`